hyperspy/drawing/figure.py

```diff
@@ -79,7 +79,8 @@ def _draw_animated(self):
             artists = sorted(ax.get_children(), key=lambda x: x.zorder)
             for artist in artists:
                 if artist.get_animated():
-                    ax.draw_artist(artist)
+                    if artist.figure is not None:  # handle race condition
```

Comment:
I'm not entirely sure why I added that but I think that removing it causes some plotting issues when operating interactively.  I think there are cases where the event loop hasn't completely run and it's possible that the artist doesn't have a figure initialized.

Reply:
I can't seem to get the bug to pop up again so we can probably remove it...

Reply:
Okay, I will remove it for now: the issue you describe sounds sensible but this is happening, I suspect that this is because the markers are not created correctly. Without it, it seems to work fine, and it is also possible that it wasn't in the course of the development of this PR and hopefully this is now fine! :)